compiler/z/codegen/OMRInstOpCode.hpp

```diff
@@ -564,28 +555,61 @@ class InstOpCode: public OMR::InstOpCode
       S390NumBranchConditions = lastBranchCondition +1
       };
 
-   struct OpCodeBinaryEntry
+   /**
+    *  \brief
+    *      Defines various metadata of an instruction including the name, description, opcodes, format, the minimum
+	 *      architecture level set (ALS) which introduced the instruction, and the various properties which model the
+	 *      instruction in a way that the code generator understands.
+    */
+   struct OpCodeMetaData
       {
-      uint8_t bytes[2];
-      uint8_t instructionFormat;
+		/**
+       *  \brief
+       *      The instruction mnemonic.
+       */
+      OMR::InstOpCode::Mnemonic mnemonic;
+
+		/**
+       *  \brief
+       *      The instruction mnemonic as defined by Principles of Operation.
+       */
+		const char* name;
+
+		/**
+       *  \brief
+       *      The fully qualified instruction name as defined by Principles of Operation.
+       */
+		const char* description;
+
+		/**
+       *  \brief
+       *      The instruction opcode which is at most two bytes in length.
+       */
+      uint8_t opcode[2];
+
+		/**
+       *  \brief
+       *      The instruction format as defined by Principles of Operation.
+       */
+      uint8_t format;
 
       /**
        *  \brief
        *      The minimum architecture level set (ALS) which introduced this instruction.
        */
       TR_S390ProcessorInfo::TR_S390ProcessorArchitectures minimumALS;
-      };
-
 
-   /* Static tables(array) that uses the OpCode as the index */
-   static const uint64_t            properties[NumOpCodes];
-   static const uint64_t            properties2[NumOpCodes];
-   static const OpCodeBinaryEntry   binaryEncodings[NumOpCodes];
-   static const char *              opCodeToNameMap[NumOpCodes];
+		/**
+       *  \brief
+       *      The properties describing the behavior of this instruction to the codegen.
+       */
+		uint64_t properties;
+      };
 
+   static const OpCodeMetaData metadata[NumOpCodes];
 
-   uint8_t getFirstByte(){return binaryEncodings[_mnemonic].bytes[0];}
-   uint8_t getSecondByte() {return binaryEncodings[_mnemonic].bytes[1];}
+   uint8_t getFirstByte(){return metadata[_mnemonic].opcode[0];}
```

Comment:
Space between `){` ?

Reply:
Actually these APIs are not even used. I've deprecated them in eb1a8f5